File: app/src/main/java/hibernate/v2/testyourandroid/helper/SensorHelper.java
package hibernate.v2.testyourandroid.helper;

import android.hardware.Sensor;

/**
 * Created by himphen on 21/5/16.
 */
public class SensorHelper {

	public static String getAccelerometerSensorData(int j, int size, String reading, Sensor sensor) {
		String[] arrayData = new String[size];
		int i = 0;
		arrayData[i] = reading;
		i++;
		arrayData[i] = sensor.getName();
		i++;
		arrayData[i] = sensor.getVendor();
		i++;
		arrayData[i] = String.valueOf(sensor.getVersion());
		i++;
		arrayData[i] = String.valueOf(sensor.getMaximumRange()) + " m/s²";
		i++;
		arrayData[i] = String.valueOf(sensor.getMinDelay()) + " μs";
		i++;
		arrayData[i] = "" + String.valueOf(sensor.getResolution()) + " m/s²";
		i++;
		arrayData[i] = String.valueOf(sensor.getPower()) + " mA";

		return arrayData[j];
	}

	public static String getGravitySensorData(int j, int size, String reading, Sensor sensor) {
		return getAccelerometerSensorData(j, size, reading, sensor);
	}

	public static String getPressureSensorData(int j, int size, String reading, Sensor sensor) {
		String[] arrayData = new String[size];
		int i = 0;
		arrayData[i] = reading;
		i++;
		arrayData[i] = sensor.getName();
		i++;
		arrayData[i] = sensor.getVendor();
		i++;
		arrayData[i] = String.valueOf(sensor.getVersion());
		i++;
		arrayData[i] = String.valueOf(sensor.getMaximumRange()) + " hPa";
		i++;
		arrayData[i] = String.valueOf(sensor.getMinDelay()) + " μs";
		i++;
		arrayData[i] = "" + String.valueOf(sensor.getResolution()) + " hPa";
		i++;
		arrayData[i] = String.valueOf(sensor.getPower()) + " mA";

		return arrayData[j];
	}

	public static String getLightSensorData(int j, int size, String reading, Sensor sensor) {
		String[] arrayData = new String[size];
		int i = 0;
		arrayData[i] = reading;
		i++;
		arrayData[i] = sensor.getName();
		i++;
		arrayData[i] = sensor.getVendor();
		i++;
		arrayData[i] = String.valueOf(sensor.getVersion());
		i++;
		arrayData[i] = String.valueOf(sensor.getMaximumRange()) + " lux";
		i++;
		arrayData[i] = String.valueOf(sensor.getMinDelay()) + " μs";
		i++;
		arrayData[i] = "" + String.valueOf(sensor.getResolution()) + " lux";
		i++;
		arrayData[i] = String.valueOf(sensor.getPower()) + " mA";

		return arrayData[j];
	}

	public static String getProximitySensorData(int j, int size, String reading, Sensor sensor) {
		String[] arrayData = new String[size];
		int i = 0;
		arrayData[i] = "";
		i++;
		arrayData[i] = sensor.getName();
		i++;
		arrayData[i] = sensor.getVendor();
		i++;
		arrayData[i] = String.valueOf(sensor.getVersion());
		i++;
		arrayData[i] = String.valueOf(sensor.getMaximumRange()) + " cm";
		i++;
		arrayData[i] = String.valueOf(sensor.getMinDelay()) + " μs";
		i++;
		arrayData[i] = "" + String.valueOf(sensor.getResolution()) + " cm";
		i++;
		arrayData[i] = String.valueOf(sensor.getPower()) + " mA";

		return arrayData[j];
	}

	public static String getMagneticSensorData(int j, int size, String reading, Sensor sensor) {
		String[] arrayData = new String[size];
		int i = 0;
		arrayData[i] = reading;
		i++;
		arrayData[i] = sensor.getName();
		i++;
		arrayData[i] = sensor.getVendor();
		i++;
		arrayData[i] = String.valueOf(sensor.getVersion());
		i++;
		arrayData[i] = String.valueOf(sensor.getMaximumRange()) + " μT";
		i++;
		arrayData[i] = String.valueOf(sensor.getMinDelay()) + " μT";
		i++;
		arrayData[i] = "" + String.valueOf(sensor.getResolution()) + " μT";
		i++;
		arrayData[i] = String.valueOf(sensor.getPower()) + " mA";

		return arrayData[j];
	}

	public static String getStepCounterSensorData(int j, int size, String reading, Sensor sensor) {
		String[] arrayData = new String[size];
		int i = 0;
		arrayData[i] = reading;
		i++;
		arrayData[i] = sensor.getName();
		i++;
		arrayData[i] = sensor.getVendor();
		i++;
		arrayData[i] = String.valueOf(sensor.getVersion());
		i++;
		arrayData[i] = String.valueOf(sensor.getMaximumRange());
		i++;
		arrayData[i] = String.valueOf(sensor.getMinDelay()) + " s";
		i++;
		arrayData[i] = "" + String.valueOf(sensor.getResolution());
		i++;
		arrayData[i] = String.valueOf(sensor.getPower()) + " mA";

		return arrayData[j];
	}

	public static String getTemperatureCounterSensorData(int j, int size, String reading, Sensor sensor) {
		String[] arrayData = new String[size];
		int i = 0;
		arrayData[i] = reading;
		i++;
		arrayData[i] = sensor.getName();
		i++;
		arrayData[i] = sensor.getVendor();
		i++;
		arrayData[i] = String.valueOf(sensor.getVersion());
		i++;
		arrayData[i] = String.valueOf(sensor.getMaximumRange());
		i++;
		arrayData[i] = String.valueOf(sensor.getMinDelay()) + " s";
		i++;
		arrayData[i] = "" + String.valueOf(sensor.getResolution());
		i++;
		arrayData[i] = String.valueOf(sensor.getPower()) + " mA";

		return arrayData[j];
	}

	public static String getHumiditySensorData(int j, int size, String reading, Sensor sensor) {
		String[] arrayData = new String[size];
		int i = 0;
		arrayData[i] = reading;
		i++;
		arrayData[i] = sensor.getName();
		i++;
		arrayData[i] = sensor.getVendor();
		i++;
		arrayData[i] = String.valueOf(sensor.getVersion());
		i++;
		arrayData[i] = String.valueOf(sensor.getMaximumRange());
		i++;
		arrayData[i] = String.valueOf(sensor.getMinDelay()) + " s";
		i++;
		arrayData[i] = "" + String.valueOf(sensor.getResolution());
		i++;
		arrayData[i] = String.valueOf(sensor.getPower()) + " mA";

		return arrayData[j];
	}
}

File: app/src/main/java/hibernate/v2/testyourandroid/helper/SensorHelper.kt
package hibernate.v2.testyourandroid.helper

import android.hardware.Sensor

/**
 * Created by himphen on 21/5/16.
 */
object SensorHelper {
    @JvmStatic
    fun getAccelerometerSensorData(j: Int, size: Int, reading: String?, sensor: Sensor): String? {
        val arrayData = arrayOfNulls<String>(size)
        var i = 0
        arrayData[i] = reading
        i++
        arrayData[i] = sensor.name
        i++
        arrayData[i] = sensor.vendor
        i++
        arrayData[i] = sensor.version.toString()
        i++
        arrayData[i] = sensor.maximumRange.toString() + " m/s²"
        i++
        arrayData[i] = sensor.minDelay.toString() + " μs"
        i++
        arrayData[i] = "" + sensor.resolution + " m/s²"
        i++
        arrayData[i] = sensor.power.toString() + " mA"
        return arrayData[j]
    }

    @JvmStatic
    fun getGravitySensorData(j: Int, size: Int, reading: String?, sensor: Sensor): String? {
        return getAccelerometerSensorData(j, size, reading, sensor)
    }

    @JvmStatic
    fun getPressureSensorData(j: Int, size: Int, reading: String?, sensor: Sensor): String? {
        val arrayData = arrayOfNulls<String>(size)
        var i = 0
        arrayData[i] = reading
        i++
        arrayData[i] = sensor.name
        i++
        arrayData[i] = sensor.vendor
        i++
        arrayData[i] = sensor.version.toString()
        i++
        arrayData[i] = sensor.maximumRange.toString() + " hPa"
        i++
        arrayData[i] = sensor.minDelay.toString() + " μs"
        i++
        arrayData[i] = "" + sensor.resolution + " hPa"
        i++
        arrayData[i] = sensor.power.toString() + " mA"
        return arrayData[j]
    }

    @JvmStatic
    fun getLightSensorData(j: Int, size: Int, reading: String?, sensor: Sensor): String? {
        val arrayData = arrayOfNulls<String>(size)
        var i = 0
        arrayData[i] = reading
        i++
        arrayData[i] = sensor.name
        i++
        arrayData[i] = sensor.vendor
        i++
        arrayData[i] = sensor.version.toString()
        i++
        arrayData[i] = sensor.maximumRange.toString() + " lux"
        i++
        arrayData[i] = sensor.minDelay.toString() + " μs"
        i++
        arrayData[i] = "" + sensor.resolution + " lux"
        i++
        arrayData[i] = sensor.power.toString() + " mA"
        return arrayData[j]
    }

    @JvmStatic
    fun getProximitySensorData(j: Int, size: Int, @Suppress("UNUSED_PARAMETER") reading: String?, sensor: Sensor): String? {
        val arrayData = arrayOfNulls<String>(size)
        var i = 0
        arrayData[i] = ""
        i++
        arrayData[i] = sensor.name
        i++
        arrayData[i] = sensor.vendor
        i++
        arrayData[i] = sensor.version.toString()
        i++
        arrayData[i] = sensor.maximumRange.toString() + " cm"
        i++
        arrayData[i] = sensor.minDelay.toString() + " μs"
        i++
        arrayData[i] = "" + sensor.resolution + " cm"
        i++
        arrayData[i] = sensor.power.toString() + " mA"
        return arrayData[j]
    }

    @JvmStatic
    fun getMagneticSensorData(j: Int, size: Int, reading: String?, sensor: Sensor): String? {
        val arrayData = arrayOfNulls<String>(size)
        var i = 0
        arrayData[i] = reading
        i++
        arrayData[i] = sensor.name
        i++
        arrayData[i] = sensor.vendor
        i++
        arrayData[i] = sensor.version.toString()
        i++
        arrayData[i] = sensor.maximumRange.toString() + " μT"
        i++
        arrayData[i] = sensor.minDelay.toString() + " μT"
        i++
        arrayData[i] = "" + sensor.resolution + " μT"
        i++
        arrayData[i] = sensor.power.toString() + " mA"
        return arrayData[j]
    }

    @JvmStatic
    fun getStepCounterSensorData(j: Int, size: Int, reading: String?, sensor: Sensor): String? {
        val arrayData = arrayOfNulls<String>(size)
        var i = 0
        arrayData[i] = reading
        i++
        arrayData[i] = sensor.name
        i++
        arrayData[i] = sensor.vendor
        i++
        arrayData[i] = sensor.version.toString()
        i++
        arrayData[i] = sensor.maximumRange.toString()
        i++
        arrayData[i] = sensor.minDelay.toString() + " s"
        i++
        arrayData[i] = "" + sensor.resolution
        i++
        arrayData[i] = sensor.power.toString() + " mA"
        return arrayData[j]
    }

    @JvmStatic
    fun getTemperatureCounterSensorData(j: Int, size: Int, reading: String?, sensor: Sensor): String? {
        val arrayData = arrayOfNulls<String>(size)
        var i = 0
        arrayData[i] = reading
        i++
        arrayData[i] = sensor.name
        i++
        arrayData[i] = sensor.vendor
        i++
        arrayData[i] = sensor.version.toString()
        i++
        arrayData[i] = sensor.maximumRange.toString()
        i++
        arrayData[i] = sensor.minDelay.toString() + " s"
        i++
        arrayData[i] = "" + sensor.resolution
        i++
        arrayData[i] = sensor.power.toString() + " mA"
        return arrayData[j]
    }

    @JvmStatic
    fun getHumiditySensorData(j: Int, size: Int, reading: String?, sensor: Sensor): String? {
        val arrayData = arrayOfNulls<String>(size)
        var i = 0
        arrayData[i] = reading
        i++
        arrayData[i] = sensor.name
        i++
        arrayData[i] = sensor.vendor
        i++
        arrayData[i] = sensor.version.toString()
        i++
        arrayData[i] = sensor.maximumRange.toString()
        i++
        arrayData[i] = sensor.minDelay.toString() + " s"
        i++
        arrayData[i] = "" + sensor.resolution
        i++
        arrayData[i] = sensor.power.toString() + " mA"
        return arrayData[j]
    }
}